### ISSUE TITLE
Crashing when closing Office 2016

### ISSUE THREAD

**Issue by tech55541** [nvaccess/nvda#7867]
Steps to Reproduce:
1. Open any program in Office 2016 (Word, Outlook, Excel, etc.).
2. Close the program.
3. NVDA stops talking and you must log off to make it respond again. Pressing CTRL+ALT+N does not start NVDA again.

Expected Behavior:
After closing Office 2016, a freeze/crash should not happen.

Actual Behavior:
A nasty freeze that's hard to make NVDA start from.

System Configuration:
- Latest version of NVDA.
- NVDA installed on local hard drive.
- JAWS sounding synthesizer installed.
- Windows 7 Professional (64-bit).
- 16GB RAM, Quad core Intel7 processor, 3.5GHZ.
- Microsoft Office Professional 2016.

Details:
I knew once upon a time this was an issue with Microsoft Office 2013. Not sure if that was ever resolved. Office 2016 has worked fine for a few months until this latest NVDA update. The update must have introduced a new bug that started the crashing cycle to begin again.

The issue still appears after restart. Previous version works fine.

Hopefully you can reproduce and release a fix.

Thanks.

**Comment by Adriani90** [comment]
I cannot reproduce this on Windows 10 Build 16299.64 with NVDA 2017.4.

However, I had such an issue in the past, reinstalling office has solved it.

**Comment by Adriani90** [comment]
@tech55541 are you still having this issue?

**Comment by Adriani90** [comment]
We do not have any feedback from the initial author since he created this issue. I am closing this for now. Please feel free to comment on it and we can reopen the discussion. Thanks!

**Comment by tech55541** [comment]
Hello,

Sorry, missed the notification. I am still on Windows 7 64-bit professional facing the same issue in Office 2016.

Thanks.

**Comment by Adriani90** [comment]
reopening. Thanks for reporting back.

**Comment by Adriani90** [comment]
Could you please upload a nvda_old.log file and a crash dump according to this article after NVDA crashes?
https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps

Thanks!

**Comment by tech55541** [comment]
Hello,

It just happened again. Here's the full log. It seems it can't find anything to focus on after exit.

`INFO - __main__ (12:14:46.673):
Starting NVDA
INFO - core.main (12:14:46.835):
Config dir: C:\Users\dell\AppData\Roaming\nvda
INFO - config.ConfigManager._loadConfig (12:14:46.835):
Loading config: C:\Users\dell\AppData\Roaming\nvda\nvda.ini
INFO - core.main (12:14:46.914):
NVDA version 2019.1.1
INFO - core.main (12:14:46.914):
Using Windows version 6.1.7601 service pack 1 workstation
INFO - core.main (12:14:46.914):
Using Python version 2.7.15 (v2.7.15:ca079a3ea3, Apr 30 2018, 16:22:17) [MSC v.1500 32 bit (Intel)]
INFO - core.main (12:14:46.914):
Using comtypes version 1.1.3
INFO - core.main (12:14:46.914):
Using configobj version 5.1.0 with validate version 1.0.1
INFO - synthDriverHandler.setSynth (12:14:47.148):
Loaded synthDriver eloquence
INFO - core.main (12:14:47.148):
Using wx version 4.0.3 msw (phoenix) wxWidgets 3.0.5
INFO - brailleInput.initialize (12:14:47.148):
Braille input initialized
INFO - braille.initialize (12:14:47.150):
Using liblouis version 3.8.0
ERROR - braille.BrailleHandler.setDisplayByName (12:14:47.157):
Error initializing display driver for kwargs {}
Traceback (most recent call last):
  File "braille.pyo", line 1658, in setDisplayByName
  File "brailleDisplayDrivers\hims.pyo", line 236, in __init__
RuntimeError: No Hims display found
INFO - braille.BrailleHandler.setDisplayByName (12:14:47.160):
Loaded braille display driver noBraille, current display has 0 cells.
WARNING - core.main (12:14:47.180):
Java Access Bridge not available
INFO - _UIAHandler.UIAHandler.MTAThreadFunc (12:14:47.184):
UIAutomation: IUIAutomation
INFO - core.main (12:14:47.644):
NVDA initialized
ERROR - eventHandler.executeEvent (12:14:48.092):
error executing event: focusEntered on <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x05578470> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyo", line 155, in executeEvent
  File "eventHandler.pyo", line 92, in __init__
  File "eventHandler.pyo", line 100, in next
  File "NVDAObjects\__init__.pyo", line 960, in event_focusEntered
  File "speech.pyo", line 403, in speakObject
  File "speech.pyo", line 277, in speakObjectProperties
  File "baseObject.pyo", line 47, in __get__
  File "baseObject.pyo", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyo", line 1148, in _get_columnNumber
  File "baseObject.pyo", line 47, in __get__
  File "baseObject.pyo", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyo", line 1127, in _get_IA2PhysicalColumnNumber
AttributeError: 'Ia2Web' object has no attribute 'IAccessibleTableObject'
ERROR - eventHandler.executeEvent (12:14:48.095):
error executing event: focusEntered on <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x05578410> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyo", line 155, in executeEvent
  File "eventHandler.pyo", line 92, in __init__
  File "eventHandler.pyo", line 100, in next
  File "NVDAObjects\__init__.pyo", line 960, in event_focusEntered
  File "speech.pyo", line 403, in speakObject
  File "speech.pyo", line 277, in speakObjectProperties
  File "baseObject.pyo", line 47, in __get__
  File "baseObject.pyo", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyo", line 1148, in _get_columnNumber
  File "baseObject.pyo", line 47, in __get__
  File "baseObject.pyo", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyo", line 1127, in _get_IA2PhysicalColumnNumber
AttributeError: 'Ia2Web' object has no attribute 'IAccessibleTableObject'
ERROR - eventHandler.executeEvent (12:14:48.099):
error executing event: focusEntered on <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x055782F0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyo", line 155, in executeEvent
  File "eventHandler.pyo", line 92, in __init__
  File "eventHandler.pyo", line 100, in next
  File "NVDAObjects\__init__.pyo", line 960, in event_focusEntered
  File "speech.pyo", line 403, in speakObject
  File "speech.pyo", line 277, in speakObjectProperties
  File "baseObject.pyo", line 47, in __get__
  File "baseObject.pyo", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyo", line 1148, in _get_columnNumber
  File "baseObject.pyo", line 47, in __get__
  File "baseObject.pyo", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyo", line 1127, in _get_IA2PhysicalColumnNumber
AttributeError: 'Ia2Web' object has no attribute 'IAccessibleTableObject'`

Thanks.

**Comment by Adriani90** [comment]
Unfortunately the log you provided has very unstructured information and we do not exactly know when occurs what. Could you please do following?

1. Got to the NVDA setting
2. In general settings, tab to logging and choose "debug"
3. Press ok and restart NVDA
4. Try to reproduce the issue and after that press NVDA+f1 if possible.
5. Save the file as .txt file and upload it to this issue

**Comment by Adriani90** [comment]
If the steps above are not working, please do following:
1. Set logging to debug as described in steps 1 to 3 above
2. Reproduce the issue
3. Press windows+r and type %temp%
4. You should find a file called "nvda-old.log"
5. Save it as .txt and upload it to this issue.
6. In %temp% there should also be a file called "nvda_crash.dmp"
7. Upload that file to this issue as well.

In this way we could investigate further where the crash comes from.

**Comment by tech55541** [comment]
Hello,

I forgot to enable debug but I was able to grab the "old" log as described. The error is so random, I really would rather not leave debug on constantly. This is what I have, please let me know if it's useful.

Thanks.
[nvda-old.log](https://github.com/nvaccess/nvda/files/3194404/nvda-old.log)

**Comment by Adriani90** [comment]
@tech55541 are you able to reproduce this now also with NVDA 2019.3 Beta 2?

**Comment by tech55541** [comment]
Hello,

I haven't been able to reproduce this in quite some time in latest NVDA version. Maybe all fixed now?

Thanks.

**Comment by Adriani90** [comment]
Thanks for reporting back. Closing as works for me.
